code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -236,7 +240,7 @@
 	if(custom_name)
 		changed_name = custom_name
 	else
-		changed_name = "[(designation ? "[designation] " : "")]Cyborg-[num2text(ident)]"
+		changed_name = "[(designation ? "[designation] " : "")][istype(mmi, /obj/item/device/mmi/posibrain) ? "Android" : "Cyborg"]-[num2text(ident)]"

```

Comment:
just give MMIs a designation variable, that defaults to "Cyborg" and is set to "Android" on Posibrains.